libfwupdplugin/fu-oprom-firmware.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2022 Intel
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuFirmware"
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-byte-array.h"
+#include "fu-bytes.h"
+#include "fu-ifwi-cpd-firmware.h"
+#include "fu-mem.h"
+#include "fu-oprom-firmware.h"
+#include "fu-string.h"
+
+/**
+ * FuOpromFirmware:
```

Comment:
I'm wondering if this should be PciOptionRomFirmware

Reply:
That's a bit wordy, but I wasn't sure so did some rough research:

optrom: 50% of qemu
oprom: 50% of EDK/uefi,coreboot
optionrom: SeaBIOS, 50% EDK, 50% optionrom, most of Linux kernel

So maybe `PciOptionRomFirmware` is best? It's going to get wordy if we split out the sections even further, e.g. `PciEfiOptionRomFirmware`. I guess my vote is Oprom just because it's shorter and what the Intel documentation seems to use but it's not that important to me either way.

Reply:
If it makes cross referencing documentation and other code easier, keep it shorter.